code/hud/hudtarget.cpp

```diff
@@ -1530,6 +1530,8 @@ void hud_target_missile(object *source_obj, int next_flag)
 				continue;
 			}
 
+			Assertion(Objects[so->objnum].type == OBJ_SHIP, "hud_target_missile was about to call obj_team with a non-ship obejct with type %d. Please report!", Objects[so->objnum].type); 
```

Comment:
And here, `continue` if not.

Reply:
This one will already continue at the next if block if it is false.

Reply:
The one starting with `		if (!iff_x_attacks_y(Player_ship->team, obj_team(A))) {`? That will then end up running `obj_team()` on it first, will it not?

Reply:
Yeah, but on release it returns -1 for invalid objects, so the check `!iff_x_attacks_y(Player_ship->team, obj_team(A)))` where they are at war aiui returns false.